File: activities/app/src/main/java/com/example/activities/MainActivity.java
package com.example.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
  TextView textView,textUrl;
 EditText editTextTextPersonName4,editTextTextPersonName3;
  Button btnadd;
  ImageView browser;
    String TAG="yash";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
      // textView=findViewById(R.id.textView);
        textUrl=findViewById(R.id.textUrl);
       editTextTextPersonName4=findViewById(R.id.editTextTextPersonName4);
        editTextTextPersonName3=findViewById(R.id.editTextTextPersonName3);
        btnadd=findViewById(R.id.btnadd);
        browser=findViewById(R.id.browser);
        btnadd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                int var1=Integer.valueOf(editTextTextPersonName3.getText().toString());
                int var2=Integer.valueOf(editTextTextPersonName4.getText().toString());
                int var3=var1+var2;
               // textView.setText(String.valueOf(var3));
                //go to second activity
                Intent i =new Intent(MainActivity.this, second.class);
                i.putExtra("result",var3);

                startActivity(i);
            }
        });
        browser.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String url=textUrl.getText().toString();
                Uri uri=Uri.parse(url);
                Intent i=new Intent(Intent.ACTION_VIEW,uri);
                try {
                    startActivity(i);
                }catch (ActivityNotFoundException exc){
                    // String TAG="yash";
                    Log.e(TAG, "onClick: not found",exc );
                    Toast.makeText(MainActivity.this, "could not found", Toast.LENGTH_SHORT).show();
                }
            }
        });

    }
}

File: tictactoe/app/src/main/java/com/example/tictactoe/MainActivity.kt
package com.example.tictactoe

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
const val KEY_1="player 1"
const val  KEY_2="player 2"
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        button11.setOnClickListener {
           val player1=editTextTextPersonName.text.toString()
            val player2=editTextTextPersonName2.text.toString()
             val intent=Intent(this,tictocgame::class.java)
            intent.putExtra(KEY_1,player1)
           intent.putExtra(KEY_2,player2)
            startActivity(intent)
        }
    }
}
